core/src/main/resources/hudson/PluginManager/check.jelly

```diff
@@ -25,9 +25,9 @@ THE SOFTWARE.
 <j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
 
   <j:invokeStatic var="ds" className="jenkins.model.DownloadSettings" method="get"/>
-  <form method="post" action="${ds.useBrowser ? 'checkUpdates' : 'checkUpdatesServer'}">
-    ${%lastUpdated(app.updateCenter.lastUpdatedString)}
+  ${%lastUpdated(app.updateCenter.lastUpdatedString)}
+  <f:link href="${ds.useBrowser ? 'checkUpdates' : 'checkUpdatesServer'}" post="true">
     <f:submit value="${%Check now}" />
```

Comment:
Something is wrong here. The only purpose of `f:link` is to install an `onclick` handler (that dynamically creates and submits an invisible `form`), which you are not even using. So it sounds like your button is submitting some _other_ containing form, with an `action` unrelated to the `href` specified here.